src/FindStmtCost.cpp

```diff
@@ -0,0 +1,930 @@
+}
+
+void FindStmtCost::visit(const GE *op) {
+    visit_binary_op(op, op->a, op->b);
+}
+
+void FindStmtCost::visit(const And *op) {
+    visit_binary_op(op, op->a, op->b);
+}
+
+void FindStmtCost::visit(const Or *op) {
+    visit_binary_op(op, op->a, op->b);
+}
+
+void FindStmtCost::visit(const Not *op) {
+    op->a.accept(this);
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, {op->a.get()});
+    set_costs(false, op, {op->a.get()});
+}
+
+void FindStmtCost::visit(const Select *op) {
+    op->condition.accept(this);
+    op->true_value.accept(this);
+    op->false_value.accept(this);
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, {op->condition.get(), op->true_value.get(), op->false_value.get()});
+    set_costs(false, op, {op->condition.get(), op->true_value.get(), op->false_value.get()});
+}
+
+void FindStmtCost::visit(const Load *op) {
+    op->predicate.accept(this);
+    op->index.accept(this);
+
+    uint8_t bits = op->type.bits();
+    uint16_t lanes = op->type.lanes();
+    int scaling_factor = get_scaling_factor(bits, lanes);
+
+    std::function<int(int)> calculate_cc = [scaling_factor](int children_cost) {
+        return scaling_factor * (NORMAL_NODE_CC + children_cost);
+    };
+
+    std::function<int(int)> calculate_dmc = [scaling_factor](int children_cost) {
+        return scaling_factor * (LOAD_DM_COST + children_cost);
+    };
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, {op->predicate.get(), op->index.get()}, calculate_cc, calculate_dmc);
+    set_costs(false, op, {op->predicate.get(), op->index.get()}, calculate_cc, calculate_dmc);
+}
+
+void FindStmtCost::visit(const Ramp *op) {
+    op->base.accept(this);
+    op->stride.accept(this);
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, {op->base.get(), op->stride.get()});
+    set_costs(false, op, {op->base.get(), op->stride.get()});
+}
+
+void FindStmtCost::visit(const Broadcast *op) {
+    op->value.accept(this);
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, {op->value.get()});
+    set_costs(false, op, {op->value.get()});
+}
+
+void FindStmtCost::visit(const Call *op) {
+    vector<const IRNode *> children;
+
+    for (const auto &arg : op->args) {
+        arg.accept(this);
+        children.push_back(arg.get());
+    }
+
+    // Consider extern call args
+    if (op->func.defined()) {
+        Function f(op->func);
+        if (op->call_type == Call::Halide && f.has_extern_definition()) {
+            for (const auto &arg : f.extern_arguments()) {
+                if (arg.is_expr()) {
+                    arg.expr.accept(this);
+                    children.push_back(arg.expr.get());
+                }
+            }
+        }
+    }
+
+    // inclusive and exclusive costs are the same
+    set_costs(true, op, children);
+    set_costs(false, op, children);
+}
+
+void FindStmtCost::visit(const Let *op) {
+
```

Comment:
Style nit, here and many other places: the first line of a function body should never be a blank line.